custom_components/tahoma/sensor.py

```diff
@@ -107,10 +125,67 @@ async def async_setup_entry(hass, entry, async_add_entities):
     entities = [
         TahomaSensor(device.deviceurl, coordinator)
         for device in data["entities"].get(SENSOR)
-        if device.states
+        if device.states and "BATTERY_" not in device.label
+    ]
+
+    battery_entities = [
+        TahomaBatterySensor(device.deviceurl, coordinator)
+        for device in data["entities"].get(SENSOR)
+        if device.states and "BATTERY_" in device.label
     ]
 
-    async_add_entities(entities)
+    async_add_entities(entities + battery_entities)
+
+
+class TahomaBatterySensor(TahomaDevice):
+    """Representation of a Tahoma device battery sensor."""
+
+    def __init__(self, device_url, coordinator):
+        """Initialize the sensor."""
+        self._battery_sensor = True
+        super().__init__(device_url, coordinator)
+
+    @property
+    def state(self):
+        """Return the value of the sensor."""
+        state = self.select_state(
+            "core:BatteryState",
+            "core:BatteryLevelState",
+            "core:SensorDefectState",
+        )
+
+        if not state:
+            return None
+        if isinstance(state, (int, float)):
+            return int(state)
+        return BATTERY_STATES[state]
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the device class of this entity if any."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return PERCENTAGE
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return f"{self.device.label} battery sensor"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self.device.deviceurl}-battery"
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device registry information for this entity."""
+        info = super().device_info
+        info["name"] = self.device.label
```

Comment:
Is this one required?

Reply:
it is as entities need to have the same device_info in order to be listed under the same device, but we change the name to add "battery" at the end.